### PR TITLE
Fix past fees

### PR DESCRIPTION
**Pull request by tromp**
[Link to rendered text](https://github.com/tromp/grin-rfcs/blob/fixpriorfees/text/0000-fix-prior-fees.md)

Implemented in PRs https://github.com/mimblewimble/grin/pull/3629 and https://github.com/mimblewimble/grin-wallet/pull/602 .

### CODE REVIEW

**Review by DavidBurkett — APPROVED** [review]

File: text/0000-fix-prior-fees.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+This makes the FeeFields methods fee() and fee\_shift() height independent,
+as well as several other functions which end up calling them.
+This results in nontrivial code and consensus model simplification,
+with no downsise, which is always a good thing.
+
+## Community-level explanation
+[community-level-explanation]: #community-level-explanation
+
+The largest fee to appear on mainnet prior to Hark Fork 4 is 2.404 Grin. This means that extending the current 40-bit fee restriction to the pre-HF4 history preserves validity.
+While this is technically a hard-forking change, it would only have any visible
```
Comment (phyro) [inline text/0000-fix-prior-fees.md:27]: this sentence "it would only have any visible" is unfinished

**Comment by phyro** [comment]
Should we move it to FCP and get it accepted ASAP so that we can also tackle the PR that implements this?

**Comment by j01tz** [comment]
In line with our governance process, this RFC can be considered being in Final Comment Period, with a disposition to merge in two weeks time, on April 20.

Please ensure any comments are made before then!

File: text/0000-fix-prior-fees.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+- RFC PR: Edit if merged: [mimblewimble/grin-rfcs#0000](https://github.com/mimblewimble/grin-rfcs/pull/0000)
+- Tracking issue: [Edit if merged with link to tracking github issue]
+
+---
+
+## Summary
+[summary]: #summary
+
+Carry the restriction of fees, to 40 bits since HF4, back to all history.
+Do not change headerversion despite the hard-forking nature.
```
Comment (quentinlesceller) [inline text/0000-fix-prior-fees.md:13]:
```suggestion
Do not change `headerversion` despite the hard-forking nature.
```
or 
```suggestion
Do not change header version despite the hard-forking nature.
```

File: text/0000-fix-prior-fees.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+## Motivation
+[motivation]: #motivation
+
+This makes the FeeFields methods fee() and fee\_shift() height independent,
```
Comment (quentinlesceller) [inline text/0000-fix-prior-fees.md:18]:
```suggestion
This makes the `FeeFields` methods `fee()` and `fee_shift()` height independent,
```

File: text/0000-fix-prior-fees.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+How many months should we wait after HF4 before rolling out this code simplification?
+I propose that 3 months is more than enough, making this RFC acceptable from mid April 2021 onwards.
+
+## Future possibilities
+[future-possibilities]: #future-possibilities
+
+## References
+[references]: #references
+
+[1] [Grin github](https://github.com/mimblewimble/grin/blob/acba73bf40242f963d8ea1e7128dfdfde6fb8853/core/src/core/transaction.rs#L181-L188)
```
Comment (quentinlesceller) [inline text/0000-fix-prior-fees.md:64]: Maybe a different link name?

**Review by quentinlesceller — APPROVED** [review]
Small format comments. Overall 👍

**Review by j01tz — APPROVED** [review]
Looks good overall, few small typos

**Pull request merged**